sled-agent/src/updates.rs

```diff
@@ -80,17 +76,19 @@ pub async fn download_artifact(
                     })
                     .await?;
             }
+            file.sync_all().await.map_err(|err| Error::Io {
+                message: "sync temp file".to_string(),
+                err,
+            })?;
```

Comment:
1. is it important to sync? This seems like it might be expensive, but I don't know how it's implemented on illumos. Is it important that it reaches persistent storage before the HTTP call returns? It seems like it's probably fine if it's still just in memory, but I don't claim to know for sure.

2. Do we need to `flush()`? It seems like we do if there might be other threads/processes/HTTP queries reading the file.

From https://docs.rs/tokio/latest/tokio/fs/struct.File.html
> A file will not be closed immediately when it goes out of scope if there are any IO operations that have not yet completed. To ensure that a file is closed immediately when it is dropped, you should call [flush](https://docs.rs/tokio/latest/tokio/io/trait.AsyncWriteExt.html#method.flush) before dropping it. Note that this does not ensure that the file has been fully written to disk; the operating system might keep the changes around in an in-memory buffer. See the [sync_all](https://docs.rs/tokio/latest/tokio/fs/struct.File.html#method.sync_all) method for telling the OS to write the data to disk.

Reply:
... I think I read that and my brain latched onto `sync_all` in the docs. will swap that out